index.md

```diff
@@ -7,5 +7,44 @@ Best practices are difficult to practice without a good environment in which
 to practice them. Here we enumerate the elements of a productive development
 environment.
 
-_Note that this is currently a placeholder; we will be adding content
+##<a name="environments"></a> Environments
+When developing and maintaining software several environments will be needed. Usually at least 4 environments are needed, each servering a different purpose:
+
+* Dev (development)
+    * Where developers have free rein(Admin) to do whatever they need to for development, such as install software for evaluation, test new architecures, or test production install instructions. 
+    * Code changes from Dev will most likely create new Int builds.
+    * Dev should be "sandboxed" from other environments to prevent any accidents to other resources.
+* Int (integration)
+    * Where developers combine code from everyone and test to make sure they merged correctly, and it works as expected. 
+    * Builds that pass in Int can be passed to Test for possible use in Prod.
+* Test
+    * Permissions and data are exact copies from Prod. 
```

Comment:
I can change the name if you like, it is just what we called things when I worked at Boeing.